src/expr/proof_node.cpp

```diff
@@ -54,8 +54,10 @@ void ProofNode::getFreeAssumptionsMap(
   std::unordered_map<ProofNode*, bool> visited;
   std::unordered_map<ProofNode*, bool>::iterator it;
   std::vector<ProofNode*> visit;
-  // the current set of formulas bound by SCOPE
-  std::unordered_set<Node, NodeHashFunction> currentScope;
+  // Maps an bound assumption to the number of bindings it is under
+  // e.g. in (SCOPE (SCOPE (ASSUME x) (x y)) (y)), y would be mapped to 2 at
+  // (ASSUME x), and x would be mapped to 1.
+  std::unordered_map<Node, uint32_t, NodeHashFunction> scopeDepth;
```

Comment:
Could you say how the scopeDepth is used in the code below, hopefully expanding the example above?

Reply:
Done in 8ee9a9a13.